CMakeLists.txt

```diff
@@ -24,6 +24,7 @@ if (USE_WASM_COMPATIBLE_SOURCES)
   # Setting the marian submodule specific cmake options for wasm
   SET(COMPILE_LIBRARY_ONLY ON CACHE BOOL "Build only the Marian library and exclude all executables.")
   SET(USE_MKL OFF CACHE BOOL "Compile with MKL support")
+  SET(USE_WASM_COMPATIBLE_MARIAN ON CACHE BOOL "Enable those marian sources that compile to wasm. Useful for debugging wasm failures by building same sources natively")
```

Comment:
Only two paths - (1) marian-dev full and (2) adapted marian targeting wasm. 

Successes at builds in (2) and additional tests for (2) are sufficient to cover the use-case here.

Reply:
> Having capability to compile "reduced" marian natively is helpful in debugging the wasm runtime failures by first reproducing the same failures on native platform with the "reduced" Marian binaries. In the past, this has proven to be useful where we could quickly diagnose a few wasm specific issues (e.g. size_t inconsistency).

Bind these to WASM tests at WASM end (if tests in (2) fail and works at (1), we inspect the boundary).